Network/Network.cpp

```diff
@@ -310,6 +321,50 @@ namespace WPEFramework
             return retval;
 	}
 
+
+        void  Network::retryIarmEventRegistration()
+        {
+            m_retryIarmEventRegistrationThread = thread(&Network::threadRetryIarmEventRegistration, this);
+
+        }
+        void  Network::threadRetryIarmEventRegistration()
+        {
+            IARM_Result_t res = IARM_RESULT_SUCCESS;
+            while(isNetworkPluginInitialized != true)
+            {
+                IARM_Result_t retVal = IARM_RESULT_SUCCESS;
+
+#ifndef NET_DISABLE_NETSRVMGR_CHECK
+                char c;
+                uint32_t retry = 0;
+                do{
+                    retVal = IARM_Bus_Call_with_IPCTimeout(IARM_BUS_NM_SRV_MGR_NAME, IARM_BUS_NETSRVMGR_API_isAvailable, (void *)&c, sizeof(c), (1000*10));
+                    if(retVal != IARM_RESULT_SUCCESS){
+                        LOGERR("retryIarmEventRegistration: NetSrvMgr is not available. Failed to activate Network Plugin, retrying");
+                        usleep(500*1000);
+                        retry++;
+                    }
+                }while((retVal != IARM_RESULT_SUCCESS) && (retry < 20));
+#endif
+
+                if(retVal != IARM_RESULT_SUCCESS)
+                {
+                    LOGERR("retryIarmEventRegistration: NetSrvMgr is not available. Failed to activate Network Plugin, retrying");
+                }
+                else
+                {
+                    IARM_CHECK( IARM_Bus_RegisterEventHandler(IARM_BUS_NM_SRV_MGR_NAME, IARM_BUS_NETWORK_MANAGER_EVENT_INTERFACE_ENABLED_STATUS, eventHandler) );
+                    IARM_CHECK( IARM_Bus_RegisterEventHandler(IARM_BUS_NM_SRV_MGR_NAME, IARM_BUS_NETWORK_MANAGER_EVENT_INTERFACE_CONNECTION_STATUS, eventHandler) );
+                    IARM_CHECK( IARM_Bus_RegisterEventHandler(IARM_BUS_NM_SRV_MGR_NAME, IARM_BUS_NETWORK_MANAGER_EVENT_INTERFACE_IPADDRESS, eventHandler) );
+                    IARM_CHECK( IARM_Bus_RegisterEventHandler(IARM_BUS_NM_SRV_MGR_NAME, IARM_BUS_NETWORK_MANAGER_EVENT_DEFAULT_INTERFACE, eventHandler) );
+                    LOGINFO("retryIarmEventRegistration: successfully subcribed to IARM event for Network Plugin");
```

Comment:
Change the logging. Add a log telemetry header as

LOGINFO
("NETWORK_AVAILABILITY_RETRY_SUCCESS: xxxxxxxxx");